src/api/cvc4cpp.h

```diff
@@ -0,0 +1,4556 @@
+     * query.
+     */
+    bool isSat () const;
+
+    /**
+     * Return true if query was an unsatisfiable checkSat() or
+     * checkSatAssuming() query.
+     */
+    bool isUnsat () const;
+
+    /**
+     * Return true if query was a checkSat() or checkSatAssuming() query and
+     * CVC4 was not able to determine (un)satisfiability.
+     */
+    bool isSatUnknown () const;
+
+    /**
+     * Return true if corresponding query was a valid checkValid() or
+     * checkValidAssuming() query.
+     */
+    bool isValid () const;
+
+    /**
+     * Return true if corresponding query was an invalid checkValid() or
+     * checkValidAssuming() query.
+     */
+    bool isInvalid () const;
+
+    /**
+     * Return true if query was a checkValid() or checkValidAssuming() query
+     * and CVC4 was not able to determine (in)validity..
+     */
+    bool isValidUnknown () const;
+
+    /**
+     * Operator overloading for equality of two results.
+     * @param r the result to compare to for equality
+     * @return true if the results are equal
+     */
+    bool operator== (const Result& r) const;
+
+    /**
+     * Operator overloading for disequality of two results.
+     * @param r the result to compare to for disequality
+     * @return true if the results are disequal
+     */
+    bool operator!= (const Result& r) const;
+
+    /**
+     * @return an explanation for an unknown query result.
+     */
+    std::string getUnknownExplanation () const;
+
+    /**
+     * @return a string representation of this result.
+     */
+    std::string toString () const;
+
+  private:
+    /* The interal result wrapped by this result. */
+    const CVC4::Result* d_result;
+};
+
+/**
+ * Serialize a result to given stream.
+ * @param out the output stream
+ * @param r the result to be serialized to the given output stream
+ * @return the output stream
+ */
+std::ostream& operator<< (std::ostream& out, const Result& r) CVC4_PUBLIC;
+
+
+/* -------------------------------------------------------------------------- */
+/* Kind                                                                       */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * The kind of a CVC4 term.
+ */
+enum CVC4_PUBLIC Kind
+{
+  /**
+   * Internal kind.
+   * Should never be exposed or created via the API.
+   */
+  INTERNAL_KIND = -2,
+  /**
+   * Undefined kind.
+   * Should never be exposed or created via the API.
+   */
+  UNDEFINED_KIND = -1,
+  /**
+   * Null kind (kind of null term Term()).
+   * Do not explicitly create via API functions other than Term().
+   */
+  NULL_EXPR,
+
+  /* Builtin --------------------------------------------------------------- */
+
+  /**
+   * Uninterpreted constant.
+   * Parameters: 2
+   *   -[1]: Sort of the constant
+   *   -[2]: ???int32_t index???
+   * Create with:
+   *   mkConst(Kind, Sort, int32_t)
+   */
+  UNINTERPRETED_CONSTANT,
+  /**
```

Comment:
@ajreynol - can you comment on the semantics of UNINTERPRETED_CONSTANT.  Should users be able to create these or are they only created internally.  Could one of these be returned as the result from getValue?

Reply:
yes, UNINTERPRETED_CONSTANT will occur in models for terms of uninterpreted sorts, but should not appear in the input (or any literal in the CNF stream for that matter).

Reply:
Uninterpreted constants are created in the TPTP parser.